src/panels/lovelace/editor/hui-dialog-edit-card.ts

```diff
@@ -53,58 +68,118 @@ export class HuiDialogEditCard extends LitElement {
     return this.shadowRoot!.querySelector("hui-yaml-card-preview")!;
   }
 
-  protected render() {
+  protected render(): TemplateResult {
     return html`
       <style>
         paper-dialog {
           width: 650px;
         }
+        .element-editor {
+          margin-bottom: 16px;
+        }
       </style>
       <paper-dialog with-backdrop>
         <h2>Card Configuration</h2>
         <paper-dialog-scrollable>
-          <hui-yaml-editor
-            .yaml="${this._cardConfig}"
-            @yaml-changed="${this._handleYamlChanged}"
-          ></hui-yaml-editor>
+          ${
+            this._editorToggle && this._elementConfig !== null
+              ? html`<div class="element-editor">${when(
+                  this._elementConfig,
+                  () => this._elementConfig,
+                  () => html`Loading...`
+                )}</div>`
+              : html`
+              <hui-yaml-editor
+                .yaml="${this._currentConfigYaml}"
+                @yaml-changed="${this._handleYamlChanged}"
+              ></hui-yaml-editor>`
+          }
           <hui-yaml-card-preview
             .hass="${this.hass}"
-            .yaml="${this._cardConfig}"
+            .yaml="${this._currentConfigYaml}"
           ></hui-yaml-card-preview>
         </paper-dialog-scrollable>
         <div class="paper-dialog-buttons">
-          <paper-button @click="${this._closeDialog}">Cancel</paper-button>
-          <paper-button @click="${this._updateConfig}">Save</paper-button>
+          <paper-button
+            @click="${this._toggleEditor}"
+          >Toggle Editor</paper-button>
+          <paper-button
+            @click="${this._closeDialog}"
+          >Cancel</paper-button>
+          <paper-button
+            @click="${this._updateConfig}"'
+          >Save</paper-button>
         </div>
       </paper-dialog>
     `;
   }
 
-  private _handleYamlChanged(ev) {
-    this._previewEl.yaml = ev.detail.yaml;
+  private _handleYamlChanged(ev: YamlChangedEvent): void {
+    this._updatePreview(ev.detail.yaml);
+  }
+
+  private _handleConfigChanged(value: LovelaceConfig): void {
+    this._currentConfigYaml = yaml.safeDump(value);
```

Comment:
I already wrote this in my previous comment and I will not allow this. This is SUPER INEFFICIENT. Every. single. key. stroke. will now result in a whole object being converted to YAML **for no reason**. To make it worse, it is converted back to JS objects inside the preview element.

We should only convert it to YAML when the users presses save. This PR cannot be merged with any other logic.

Reply:
```
We *should* convert every change to YAML. Do less and be lazy: we only need it as YAML when the user presses thet save button. We should store the exact same format as that we pass the preview element.
```
Sorry I was a bit confused... I should have asked for clarification. Will make the change in the PR

Reply:
🤦‍♂️  damn it. Sorry about that.